### PR TITLE
Migrate Google Fonts main library to NNBD

### PR DESCRIPTION
**Pull request by pennzht**
* This PR migrates the `lib/` directory of Google Fonts to NNBD, partially unblocking downstream packages.
* One file (`application_support_directory.dart`) is **not migrated yet**; it depends on the upstream package `path_provider` which has not been migrated yet.
    + This file isolates one method that is not migrated yet (`getApplicationSupportDirectory`). This is necessary because it allows us to check the null-safety of all other files.
* Most tests are not migrated yet, but all tests are passing.
* The code for the generator itself is not migrated, **but the generator generates null-safe code.**
    + The new `google_fonts.dart` is generated by the generator and passes `dart analyze .`.
* This PR also migrates the example (`example/`) to NNBD.
* In order to enable NNBD testing on CI, we configured the CI to use `beta` channel instead of `stable`.

Related issues:
* https://github.com/material-foundation/google-fonts-flutter/issues/135
* https://github.com/material-foundation/google-fonts-flutter/issues/136

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -2,24 +2,35 @@ name: google_fonts
 description: A package to include fonts from fonts.google.com in your Flutter app.
 version: 1.1.1
```
Comment (guidezpl) [inline pubspec.yaml:3]: version should be `2.0.0-nullsafety` or `2.0.0-nullsafety.0`

File: example/lib/main.dart [hunk 2]
```diff
@@ -36,7 +36,7 @@ class _MyHomePageState extends State<MyHomePage> {
 
   @override
   Widget build(BuildContext context) {
-    final TextStyle display1 = Theme.of(context).textTheme.headline4;
+    final TextStyle display1 = Theme.of(context).textTheme.headline4!;
```
Comment (guidezpl) [inline example/lib/main.dart:39]: nit: rename display1 to headline4

**Review by guidezpl — APPROVED** [review]
```
  _    ___ _____ __  __ 
 | |  / __|_   _|  \/  |
 | |_| (_ | | | | |\/| |
 |____\___| |_| |_|  |_|
                        
```

File: example/lib/main.dart (hunk 2 above)
Comment (pennzht) [reply example/lib/main.dart]: Done.

File: pubspec.yaml (hunk 1 above)
Comment (pennzht) [reply pubspec.yaml]: Done.

File: pubspec.yaml [hunk 3]
```diff
@@ -2,24 +2,35 @@ name: google_fonts
...
 
 environment:
-  sdk: ">=2.3.0 <3.0.0"
+  sdk: ">=2.12.0-0 <3.0.0"
   flutter: ^1.17.0
 
 dependencies:
+  # The dependencies `http` and `crypto` are migrated to NNBD, but not
+  # published yet.
+  # Once the NNBD versions for these packages are published on pub.dev,
```
[the remark this answers is not in the document]
Comment (pennzht) [reply pubspec.yaml:14]: Done.

File: lib/src/application_support_directory.dart [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// @dart=2.9
```
Comment (perclasson) [inline lib/src/application_support_directory.dart:1]: I think you want to ignore this file as it gives an analyzer error:

``// ignore_for_file: import_of_legacy_library_into_null_safe
``

**Review by perclasson — APPROVED** [review]
LGTM

File: .github/workflows/actions.yml [hunk 5]
```diff
@@ -17,7 +17,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           repository: 'flutter/flutter'
-          ref: 'stable'
+          ref: 'beta'
```
Comment (perclasson) [inline .github/workflows/actions.yml:20]: I noticed now too that the name of this step needs to be renamed to:

`Clone Flutter repository with beta channel`

Or maybe just:

`Clone Flutter repository`

Comment (pennzht) [reply .github/workflows/actions.yml:20]: Done. Nice catch!

File: lib/src/application_support_directory.dart (hunk 4 above)
Comment (pennzht) [reply lib/src/application_support_directory.dart]: Thank you! It works!

File: lib/src/file_io_desktop_and_mobile.dart [hunk 6]
```diff
@@ -25,7 +25,7 @@ Future<ByteData> loadFontFromDeviceFileSystem(String name) async {
 }
 
 Future<String> get _localPath async {
-  final directory = await path_provider.getApplicationSupportDirectory();
+  final directory = await getApplicationSupportDirectory();
```
Comment (johnsonmh) [inline lib/src/file_io_desktop_and_mobile.dart:28]: nit: when calling top-level functions that are defined in other files, I prefer to add an import prefix so I can tell which import the function is coming from.


something like

Above:
```dart
import 'application_support_directory.dart' as local_directory;
```

Here:
```dart
final directory = await local_directory.getApplicationSupportDirectory();
```

**Review by johnsonmh — APPROVED** [review]
```
██╗      ██████╗ ████████╗███╗   ███╗
██║     ██╔════╝ ╚══██╔══╝████╗ ████║
██║     ██║  ███╗   ██║   ██╔████╔██║
██║     ██║   ██║   ██║   ██║╚██╔╝██║
███████╗╚██████╔╝   ██║   ██║ ╚═╝ ██║
╚══════╝ ╚═════╝    ╚═╝   ╚═╝     ╚═╝
```

File: lib/src/file_io_desktop_and_mobile.dart (hunk 6 above)
Comment (pennzht) [reply lib/src/file_io_desktop_and_mobile.dart]: Done. Thank you!

**Pull request merged**